### PR TITLE
Multi agent behaviour

### PR DESCRIPTION
**Pull request by pgavigan**
Merge the multi agent behaviour branch. Contains some code cleanup as well as the beginning of a multi-agent behaviour implementation.

### CODE REVIEW

File: saviSimulation/savi/asl/demo.asl [hunk 1]
```diff
@@ -6,3 +6,4 @@
 
 //{include("lostInThought.asl")}
 {include("working.asl")}
+//{include("multiAgent.asl")}
```
Comment (ntartania) [inline saviSimulation/savi/asl/demo.asl]: this being commented out might explain why Cristina didn't see any changes in the behaviour

Comment (cruizm) [reply saviSimulation/savi/asl/demo.asl]: I uncommented it, and there is anything that I can appreciate from the behavior

Comment (ntartania) [reply saviSimulation/savi/asl/demo.asl]: isn't anything?

Comment (cruizm) [reply saviSimulation/savi/asl/demo.asl]: Looking at simulation "video", we do not see anything different.

Comment (pgavigan) [reply saviSimulation/savi/asl/demo.asl]: There is some code cleanup as well as the beginning of some functionality needed for multi agent. On the surface, you won't be able to observe much. The changes are behind the scenes, and more of interest to @MikeVezina .

**Pull request merged**